boltz/liquid.go

```diff
@@ -182,26 +159,59 @@ func constructLiquidTransaction(network *Network, outputs []OutputDetails, outpu
 		return nil, errors.New("Failed to unblind inputs: " + err.Error())
 	}
 
-	var inputSum uint64
-	for _, input := range ownedInputs {
-		inputSum += input.Value
+	outValues := make(map[string]uint64)
+	for i, input := range ownedInputs {
+		address := outputs[i].Address
+		//nolint:gosimple
+		existingValue, _ := outValues[address]
+		outValues[address] = existingValue + input.Value
 	}
 
 	btcAsset := network.Liquid.AssetID
 
-	if err := updater.AddOutputs([]psetv2.OutputArgs{
-		{
-			Asset:        btcAsset,
-			Amount:       inputSum - fee,
-			Script:       script,
-			BlindingKey:  blindingKeyCompressed,
-			BlinderIndex: 0,
-		},
+	var blindingKeyCompressed []byte
```

Comment:
Should be defined in the loop where it is being used

Reply:
Its required after the loop aswell